src/Core/Style.js

```diff
@@ -692,14 +692,16 @@ class Style {
             this.fill.color = color;
             this.fill.opacity = readVectorProperty(layer.paint['fill-opacity']) || opacity;
             if (layer.paint['fill-pattern']) {
-                if (!sprites) {
-                    console.warn("argument sprites must not be null when using layer.paint['fill-pattern']");
+                try {
+                    this.fill.pattern = {
+                        id: layer.paint['fill-pattern'],
+                        source: sprites.source,
+                        cropValues: sprites[layer.paint['fill-pattern']],
+                    };
+                } catch (err) {
+                    err.message = "argument sprites must not be null when using layer.paint['fill-pattern']";
```

Comment:
ok :+1: can you add the layer id in the message please so it is easier to find the culprit? (same for the other try catch block)

Reply:
I add the Vector Tile layer id information on the message. We do'tn have acces to the itowns.layer as this fucntion is called by the VectorTileSource.